platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java

```diff
@@ -0,0 +1,364 @@
+                SITE_NAME_DESCRIPTION, null));
+        properties.add(getSystemPropertyDetails(SystemInfo.VERSION, VERSION_TITLE,
+                VERSION_DESCRIPTION, null));
+
+        return properties;
+    }
+
+    @Override
+    public void writeSystemProperties(Map<String, String> updatedSystemProperties) {
+        if (updatedSystemProperties == null) {
+            return;
+        }
+        // Get system.properties file
+        //save off the current/old hostname before we make any changes
+        oldHostName = baseUrl.getHost();
+
+        String etcDir = System.getProperty(KARAF_ETC);
+        String systemPropertyFilename = etcDir + File.separator + SYSTEM_PROPERTIES_FILE;
+        String userPropertiesFilename = etcDir + File.separator + USERS_PROPERTIES_FILE;
+
+        File systemPropertiesFile = new File(systemPropertyFilename);
+        File userPropertiesFile = new File(userPropertiesFilename);
+
+        try {
+            Properties systemDotProperties = new Properties(systemPropertiesFile);
+
+            updateProperty(SystemBaseUrl.PORT, updatedSystemProperties, systemDotProperties);
+            updateProperty(SystemBaseUrl.HOST, updatedSystemProperties, systemDotProperties);
+            updateProperty(SystemBaseUrl.PROTOCOL, updatedSystemProperties, systemDotProperties);
+            updateProperty(SystemBaseUrl.HTTP_PORT, updatedSystemProperties, systemDotProperties);
+            updateProperty(SystemBaseUrl.HTTPS_PORT, updatedSystemProperties, systemDotProperties);
+            updateProperty(SystemInfo.ORGANIZATION, updatedSystemProperties, systemDotProperties);
+            updateProperty(SystemInfo.SITE_CONTACT, updatedSystemProperties, systemDotProperties);
+            updateProperty(SystemInfo.SITE_NAME, updatedSystemProperties, systemDotProperties);
+            updateProperty(SystemInfo.VERSION, updatedSystemProperties, systemDotProperties);
+
+            systemDotProperties.save();
+
+        } catch (IOException e) {
+            LOGGER.warn("Exception while writing to system.properties file.", e);
+        }
+
+        try {
+            Properties userDotProperties = new Properties(userPropertiesFile);
+
+            if (!userDotProperties.isEmpty()) {
+                String oldHostValue = userDotProperties.getProperty(oldHostName);
+
+                if (oldHostValue != null) {
+                    userDotProperties.remove(oldHostName);
+                    userDotProperties
+                            .setProperty(System.getProperty(SystemBaseUrl.HOST), oldHostValue);
+
+                    userDotProperties.save();
+                }
+
+            }
+
+        } catch (IOException e) {
+            LOGGER.warn("Exception while writing to users.properties file.", e);
+
+        }
+
+    }
+
+    private SystemPropertyDetails getSystemPropertyDetails(String key, String title,
+            String description, List<String> options) {
+        String defaultValue = "";
```

Comment:
I'm a little confused as to why defaultValue and value are separate variables when they end up getting the same value. I understand that in the constructor they are different, just a little confused here.